src/main/java/com/fasterxml/jackson/module/paramnames/ParameterNamesAnnotationIntrospector.java

```diff
@@ -14,15 +15,16 @@
  * @see AnnotationIntrospector
  * @see Parameter
  */
-class ParameterNamesAnnotationIntrospector extends NopAnnotationIntrospector
-{
+class ParameterNamesAnnotationIntrospector extends NopAnnotationIntrospector {
     private static final long serialVersionUID = 1L;
 
     private final JsonCreator.Mode creatorBinding;
+    private final ParameterExtractor parameterExtractor;
 
-    ParameterNamesAnnotationIntrospector(JsonCreator.Mode creatorBinding) {
+    ParameterNamesAnnotationIntrospector(JsonCreator.Mode creatorBinding, ParameterExtractor parameterExtractor) {
```

Comment:
For backwards compatibility, would it make sense to keep old one to pass new one with default of `ParameterExtractor`?